module.c

```diff
@@ -442,6 +505,87 @@ static PyObject *context_gc(ContextData *self) {
 	Py_RETURN_NONE;
 }
 
+static JSValue js_c_function(
+    JSContext *ctx, JSValueConst this_val, int argc, JSValueConst *argv, int magic) {
+	ContextData *self = (ContextData *)JS_GetContextOpaque(ctx);
+	PythonCallableNode *node = self->python_callables;
+	while (node && node->magic != magic) {
+		node = node->next;
+	}
+	if (!node) {
+		return JS_ThrowInternalError(self->context, "Internal error.");
+	}
+	prepare_call_python(self);
+
+	PyObject *args = PyTuple_New(argc);
+	if (!args) {
+		end_call_python(self);
+		return JS_ThrowOutOfMemory(self->context);
+	}
+	for (int i = 0; i < argc; ++i) {
+		PyObject *arg = quickjs_to_python(self, JS_DupValue(self->context, argv[i]));
+		PyTuple_SET_ITEM(args, i, arg);
+	}
+
+	PyObject *result = PyObject_CallObject(node->obj, args);
+	Py_DECREF(args);
+	if (!result) {
+		end_call_python(self);
+		return JS_ThrowInternalError(self->context, "Python call failed.");
+	}
+	JSValue js_result = JS_NULL;
+	if (python_to_quickjs_possible(self, result)) {
+		js_result = python_to_quickjs(self, result);
+	}
+	Py_DECREF(result);
+
+	end_call_python(self);
+	return js_result;
+}
+
+static PyObject *context_add_callable(ContextData *self, PyObject *args) {
+	const char *name;
+	PyObject *callable;
+	if (!PyArg_ParseTuple(args, "sO", &name, &callable)) {
+		return NULL;
+	}
+	if (!PyCallable_Check(callable)) {
+		PyErr_SetString(PyExc_TypeError, "Argument must be callable.");
+		return NULL;
+	}
+
+	PythonCallableNode *node = PyMem_Malloc(sizeof(PythonCallableNode));
+	if (!node) {
+		return NULL;
+	}
+	Py_INCREF(callable);
+	node->magic = 0;
+	if (self->python_callables) {
+		node->magic = self->python_callables->magic + 1;
+	}
+	node->obj = callable;
+	node->next = self->python_callables;
+	self->python_callables = node;
+
+	JSValue function = JS_NewCFunctionMagic(
+	    self->context,
+	    js_c_function,
+	    "python",  // TODO: Name of the function should not matter that much?
+	    1,  // TODO: Should we allow setting the .length of the function to something other than 1?
+	    JS_CFUNC_generic_magic,
+	    node->magic);
+	JSValue global = JS_GetGlobalObject(self->context);
+	// If this fails we don't notify the caller of this function.
```

Comment:
I don't understand this comment

Reply:
This was added based on your previous comment 

> JS_DefinePropertyStr would technically fit better, but this is fine. Add a comment about dropping exceptions from this on the floor

Happy to take suggestions.

Reply:
I see, I think I wrote that comment based on an earlier version of this which didn't look at the return value of JS_SetPropertyStr. Or I might be misremembering. Either way it can be dropped.